grails-app/services/org/pih/warehouse/product/ProductCatalogService.groovy

```diff
@@ -17,6 +19,10 @@ class ProductCatalogService {
 
         if (params.productCode) {
             product = Product.findByProductCode(params.productCode)
+
+            if (!product) {
+                throw new ValidationException("Could not locate product with code: " + params.productCode, null);
```

Comment:
Not sure if we can pass null here. If we can, then I guess it's fine. Otherwise it would be better to use a different exception like IllegalArgumentException or IllegalStateException.

One other issue is that if params.productCode is empty this will also lead to a weird exception like the one we saw today. So you might want to pull this out of that code block. In fact I would check to see if there's a valid product AND product catalog just before we try to find the product catalog item on line 32.

Reply:
Changed, now product and product catalog code are checked even if codes are empty